packages/rum-core/src/domain/view/viewMetrics/trackScrollMetrics.ts

```diff
@@ -18,38 +28,79 @@ export function trackScrollMetrics(
   configuration: RumConfiguration,
   viewStart: ClocksState,
   callback: (scrollMetrics: ScrollMetrics) => void,
-  getScrollValues = computeScrollValues
+  getScrollValues = computeScrollValues,
+  throttleDuration: number = THROTTLE_SCROLL_DURATION
 ) {
-  let maxDepth = 0
-  const handleScrollEvent = throttle(
-    () => {
-      const { scrollHeight, scrollDepth, scrollTop } = getScrollValues()
+  let maxScrollDepth = 0
+  let maxScrollHeight = 0
+  let maxScrollTime = 0 as Duration
 
-      if (scrollDepth > maxDepth) {
-        const now = relativeNow()
-        const maxDepthTime = elapsed(viewStart.relative, now)
-        maxDepth = scrollDepth
-        callback({
-          maxDepth,
-          maxDepthScrollHeight: scrollHeight,
-          maxDepthTime,
-          maxDepthScrollTop: scrollTop,
-        })
-      }
-    },
-    THROTTLE_SCROLL_DURATION,
-    { leading: false, trailing: true }
-  )
+  let stop: () => void
+
+  const updateScrollMetrics = (height = 0) => {
+    let shouldUpdate = false
+    const { scrollDepth, scrollTop } = getScrollValues()
+    if (scrollDepth > maxScrollDepth) {
+      maxScrollDepth = scrollDepth
+      shouldUpdate = true
+    }
+
+    if (height > maxScrollHeight) {
+      maxScrollHeight = height
+      const now = relativeNow()
+      maxScrollTime = elapsed(viewStart.relative, now)
+      shouldUpdate = true
+    }
+
+    if (shouldUpdate) {
+      callback({
+        maxDepth: maxScrollDepth,
+        // TODO: This should be renamed to maxScrollHeight in the next major release
+        maxDepthScrollHeight: maxScrollHeight,
+        // TODO: This should be renamed to maxScrollTime in the next major release
+        maxDepthTime: maxScrollTime,
+        maxDepthScrollTop: scrollTop,
+      })
+    }
+  }
+
+  if (window.ResizeObserver) {
+    const trotthledUpdateScrollMetrics = throttle(updateScrollMetrics, throttleDuration, {
+      leading: false,
+      trailing: true,
+    })
 
-  const { stop } = addEventListener(configuration, window, DOM_EVENT.SCROLL, handleScrollEvent.throttled, {
-    passive: true,
-  })
+    const observerTarget = document.scrollingElement || document.documentElement
+    const resizeObserver = new ResizeObserver(
+      monitor((entries) => {
+        const height = Math.round(entries?.[0]?.borderBoxSize?.[0]?.blockSize ?? 0)
+        if (height > 0) {
```

Comment:
#
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Why this check?